client/src/components/PrivateRoute.js

```diff
@@ -0,0 +1,32 @@
+import React, { useEffect, useState } from "react";
+import { Route, Redirect } from "react-router-dom";
+
+const axios = require("axios");
+
+const PrivateRoute = ({ component: Component, ...rest }) => {
+  const [isLoaded, setIsLoaded] = useState(false);
+  const [isAuthenticated, setIsAuthenticated] = useState(false);
+
+  useEffect(() => {
+    axios
+      .get("/profile")
+      .then((res) => {
+        setIsAuthenticated(true);
+      })
+      .catch((err) => {})
+      .finally(() => setIsLoaded(true));
+  }, [isAuthenticated]);
+
+  if (!isLoaded) return <></>;
```

Comment:
This component is great!
My only question is, do we have to hit /profile every time we want to load a private route? Is it possible to check if the user was already authenticated from before?

Reply:
With the current implementation, the /profile route is used to ensure the "token" cookie which holds the JWT is still valid (it also contains an expiry date). As the cookie data cannot be accessed client-side, I am not entirely sure if there is a less resource-intensive way to check for valid authentication on every private route...